### PR TITLE
#45 Add register_table/deregister_table and expose some public mod

### PR DESCRIPTION
**Pull request by jychen7**
from https://github.com/datafusion-contrib/datafusion-python/issues/45
prerequisite for https://github.com/datafusion-contrib/datafusion-bigtable/pull/3

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -38,8 +38,8 @@ uuid = { version = "0.8", features = ["v4"] }
 mimalloc = { version = "*", default-features = false }
 
 [lib]
-name = "_internal"
-crate-type = ["cdylib"]
+name = "datafusion_python"
```
Comment (jychen7) [inline Cargo.toml:41]: want to reference in https://github.com/datafusion-contrib/datafusion-bigtable/pull/3/files#diff-775764fdcd55c2e241ba308b95a9f1f162d892a81ed83eab7d77fc2a0d09c948

File: datafusion/tests/test_catalog.py [hunk 2]
```diff
@@ -18,38 +18,6 @@
 import pyarrow as pa
 import pytest
 
-from datafusion import ExecutionContext
-
-
-@pytest.fixture
```
Comment (jychen7) [inline datafusion/tests/test_catalog.py:24]: moved to datafusion/tests/conftest.py for shared usage in `test_context` and `test_sql`

File: src/catalog.rs [hunk 3]
```diff
@@ -38,7 +38,7 @@ pub(crate) struct PyDatabase {
 }
 
 #[pyclass(name = "Table", module = "datafusion", subclass)]
-pub(crate) struct PyTable {
+pub struct PyTable {
```
Comment (jychen7) [inline src/catalog.rs:41]: want to do pyo3 inheritance in https://github.com/datafusion-contrib/datafusion-bigtable/pull/3/files#diff-775764fdcd55c2e241ba308b95a9f1f162d892a81ed83eab7d77fc2a0d09c948

File: src/utils.rs [hunk 4]
```diff
@@ -25,7 +25,7 @@ use datafusion::physical_plan::functions::Volatility;
 use crate::errors::DataFusionError;
 
 /// Utility to collect rust futures with GIL released
-pub(crate) fn wait_for_future<F: Future>(py: Python, f: F) -> F::Output
+pub fn wait_for_future<F: Future>(py: Python, f: F) -> F::Output
```
Comment (jychen7) [inline src/utils.rs:28]: want to reuse `wait_for_future` in https://github.com/datafusion-contrib/datafusion-bigtable/pull/3/files#diff-775764fdcd55c2e241ba308b95a9f1f162d892a81ed83eab7d77fc2a0d09c948

**Comment by jychen7** [comment]
@Jimexist , see if you have time to take a look

also cc @matthewmturner (from Q2 plan, I saw you have interest in python binding as well)

**Review by Jimexist — APPROVED** [review]

**Pull request merged**